### PR TITLE
Form updates: navigation buttons, progress bar, conditionally display part 2 questions based on role

### PR DESCRIPTION
**Pull request by panjenny0**
### Form updates
- wrapped the forms with their own navigation buttons and progress bars #185 
- added example navigation handlers (should include force save and make the function async when using autosave)
- fixed some mobile accessibility issues with form container and spacing

#### note:
when implementing autosave for the forms, the `autosaveTime` should be updated using `application.submission.lastUpdated`

### Other changes
- added changes to address #122 
- fixed dropdown issue where saved values werent getting displayed fyi @shuting-hu @Ciuffi 

ezgif com-gif-maker

### todo if we have time
incrementally fill progress bar % after every completed input field #191

### CODE REVIEW

**Review by vincentchiang1 — APPROVED** [review]

**Review by kevin-zou — COMMENTED** [review]
Checking for my personal understanding here, but everything looks great! 🔥

File: src/containers/Application/Part2.js [hunk 1]
```diff
@@ -11,9 +16,28 @@ export default () => {
     longAnswer: '',
   })
 
+  // https://github.com/nwplus/livesite/pull/190/files
+  const handleNavigation = href => {
+    // await forceSave()  ** add async when forceSave() is used **
```
[the remark this answers is not in the document]
Comment (panjenny0) [reply src/containers/Application/Part2.js:21]: Good question! I just wanted to add them as templates for parts 2, 3, and the review page to use, but it's up to the individual implementing the page to comment out the forceSave part and decide how to use them!

File: src/containers/Application/Part2.js [hunk 2]
```diff
@@ -11,9 +16,28 @@ export default () => {
...
+        onChange={setStates}
+        role={application.basicInfo.contributionRole}
+      />
+      <VerticalProgressBar percent={50} />
+      <NavigationButtons
+        firstButtonText="Back"
+        firstButtonOnClick={() => handleNavigation('/application/part-1')}
+        secondButtonText="Next"
+        secondButtonOnClick={() => handleNavigation('/application/part-3')}
+        autosaveTime="4:20am" // TODO: replace with application.submission.lastUpdated.toDate().toString()
```
[the remark this answers is not in the document]
Comment (panjenny0) [reply src/containers/Application/Part2.js:39]: oh it's ok! I just added a temporary placeholder for the time, but when autosave is actually being incorporated, whoever's implementing the page will just need to replace 4:20 with `application.submission.lastUpdated.toDate().toString()`

File: src/containers/Application/Part2.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (kevin-zou) [reply src/containers/Application/Part2.js:21]: oohhh gotcha! thanks for the clarification 👏

File: src/containers/Application/Part2.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (kevin-zou) [reply src/containers/Application/Part2.js:39]: sounds good! thanks for clarifying 🔥

**Review by kevin-zou — APPROVED** [review]
Lgtm! :shipit:

**Review by andykwan1124 — DISMISSED** [review]
lgtm, nice catch on the value in the dropdown component!

**Review by andykwan1124 — APPROVED** [review]

**Pull request merged**